src/Simulation/DepletionVoltage.jl

```diff
@@ -21,155 +21,138 @@ end
+    grid_size = size(simDV.point_types.grid)
+    depletion_voltage_field = zeros(T, grid_size)
+
+    _estimate_depletion_voltage_factor!(
+        depletion_voltage_field, 
+        grid_size, 
+        simDV.point_types, 
+        simDV.electric_potential.data, 
+        simDV.weighting_potentials[contact_id].data 
+    )
+
+    depletion_voltage = maximum(depletion_voltage_field) * u"V"
+
+    # estimated_uncertainty = fss.refinement_limits[end] * depletion_voltage
+    return depletion_voltage
+end
+
+
+function _estimate_depletion_voltage_factor(
+        center_imp_value::T, 
+        center_zero_imp_value::T, 
+        neighbor_imp_values::AbstractVector{T}, 
+        neighbor_zero_imp_values::AbstractVector{T}
+    ) where {T}
+    estimation = abs(center_imp_value) / center_zero_imp_value 
+    neighbor_values = neighbor_zero_imp_values .* estimation .+ neighbor_imp_values
+    vmin, vmax = extrema(neighbor_values)
+    center_value = center_zero_imp_value * estimation + center_imp_value
+    if center_value < vmin
+        estimation += (vmin - center_value) / center_zero_imp_value 
+    elseif center_value > vmax
+        estimation -= (vmax - center_value) / center_zero_imp_value
     end
-    
-    scale = 0
-    undepleted = 0
-    eps = 1e-1
-    length_idx = length(inside[1])
-    @showprogress for (i,idx) in enumerate(inside)
-        for scale_loc in local_range
-            center_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[idx] + ϕρ[idx]
-            min_pot = center_pot + eps
-            max_pot = center_pot - eps
-            for neighbour in neighbours
-                n_idx = idx + neighbour
-                if all([!(n_idx[j] == 0 || n_idx[j] > size_data[j]) for j in 1:length_idx]) # Check whether one of the indices is outside of the grid
-                    local_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[n_idx] + ϕρ[n_idx]
-                    if local_pot < min_pot
-                        min_pot = local_pot
-                    elseif local_pot > max_pot
-                        max_pot = local_pot
+    # neighbor_values = neighbor_zero_imp_values .* estimation .+ neighbor_imp_values
+    # center_value = center_zero_imp_value * estimation + center_imp_value
+    # vmin, vmax = extrema(neighbor_values)
+    # @assert (center_value < vmin || center_value > vmax) 
+    return estimation
+end
+
+function _replace_NaN_with_minimum(v::AbstractVector)
+    min = minimum(filter(x -> !isnan(x), v))
+    map(x -> isnan(x) ? min : x, v)
+end
+
+function _estimate_depletion_voltage_factor!(
+        depletion_voltage_field::AbstractArray{T}, 
+        grid_size, 
+        pts, 
+        imp_field, 
+        zero_imp_field
+    ) where {T}
+    @inbounds begin
+        for i3 in 1:grid_size[3]
+            for i2 in 1:grid_size[2]
+                for i1 in 1:grid_size[1]
+                    if SolidStateDetectors.is_pn_junction_point_type(pts[i1, i2, i3])
```

Comment:
avoid prefix `SolidStateDetectors` in the package (not only here)

Reply:
Ups, sorry. Remaining of Copy&Paste from testing script.